infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/KubernetesPluginsToolingApplier.java

```diff
@@ -180,8 +192,84 @@ private void addSidecar(
     InternalMachineConfig machineConfig = machineResolver.resolve();
     kubernetesEnvironment.getMachines().put(machineName, machineConfig);
 
+    sidecarRelatedCommands.forEach(c -> c.getAttributes().put("machineName", machineName));
+
+    container
+        .getCommands()
+        .stream()
+        .map(c -> asCommand(machineName, c))
+        .forEach(c -> kubernetesEnvironment.getCommands().add(c));
+
     SidecarServicesProvisioner sidecarServicesProvisioner =
         new SidecarServicesProvisioner(containerEndpoints, pod.getMetadata().getName());
     sidecarServicesProvisioner.provision(kubernetesEnvironment);
   }
+
+  private CommandImpl asCommand(String machineName, Command command) {
+    CommandImpl cmd =
+        new CommandImpl(
+            command.getName(),
+            command.getCommand().stream().collect(Collectors.joining(" ")),
+            "custom");
+    cmd.getAttributes().put("workDir", command.getWorkingDir());
+    cmd.getAttributes().put("machineName", machineName);
+    return cmd;
+  }
+
+  private static class CommandsResolver {
```

Comment:
Why not move this class outside of KubernetesPluginsToolingApplier? It is already a bit bloated.

Reply:
It is quite small and used only by `KubernetesPluginsToolingApplier`. I would leave it as an internal class.